File: sdk/typescript/examples/anthropic_claude.ts
import { memoryOSApiKey, memoryOSBaseUrl } from "./env.js";
import { MemoryOS } from "../src/index.js";

async function main(): Promise<void> {
  const client = new MemoryOS(memoryOSApiKey(), memoryOSBaseUrl());
  const memories = await client.get({
    query: "customer support style",
    externalUserId: "student_44821",
    limit: 4,
    format: "xml",
    contextMaxTokens: 300,
  });
  const xmlBlock = memories.hasContext ? memories.systemPromptAddition : "<memories />";
  console.log("Claude system prompt context:");
  console.log(xmlBlock);
}

void main();

File: sdk/python/examples/anthropic_claude.py
import os

from memoryos import Memory


def main() -> None:
    client = Memory(
        api_key=os.environ["MEMORYOS_API_KEY"],
        base_url=os.getenv("MEMORYOS_BASE_URL", "http://127.0.0.1:8000"),
    )
    memories = client.get(
        query="customer support style",
        external_user_id="student_44821",
        limit=4,
        format="xml",
        context_max_tokens=300,
    )
    xml_block = memories.system_prompt_addition if memories.has_context else "<memories />"
    print("Claude system prompt context:")
    print(xml_block)


if __name__ == "__main__":
    main()
